n2y/blocks.py

```diff
@@ -39,6 +39,7 @@ def __init__(self, client, notion_data, page=None, get_children=True):
         self.client = client
         self.page = page
 
+        self.notion_block = notion_data
```

Comment:
`self.notion_block = notion_data` whereas `self.notion_data = notion_data[notion_data['type']]`.  I think renaming what is currently self.notion_data to self.notion_block and vice versa would more accurate to what the api calls them but they do have different values. I wanted to start saving the original dictionary that comes in because depending on the block type, sometimes information is lost or mutated when parsing it from key-values to class attributes. Thus, instead of having to reconstruct the class instantiation's `notion_block` form by creating a dictionary or mutating `block.__dict__` (which is sometimes impossible due to the lossy process I mentioned earlier), I figured we can just save it when it comes in.

Reply:
@sHermanGriffiths my main concern is that, although they have different values, `notion_data[notion_data['type']]` can be derived from just `notion_data`. Thus, storing both of them is redundant. I hadn't thought we needed all of the content of `notion_data`, which is why previously we used `notion_data[notion_data['type']]`. However, now that we do need everything in `notion_data`, I think we can just always save everything.

Reply:
sorry I misunderstood.  Would you still be okay with saving the value of `notion_data[notion_data['type']] ` as `block.notion_block`? I only ask because otherwise, we'd be replacing the use of `block.notion_data` with `block.notion_data[block.notion_type]`. I still think it would be worthwhile to save such a wordy and widely used value as a class attribute.

Reply:
It just feels bad to have duplicate copies of the data. How about this:

```
@property
def notion_type_data(self):
  return self.notion_data[self.notion_data["type"]]
```

This avoids duplicating the data or the code for pulling out the Notion type data. Note you'll still need to refactor the naming a bit.

Reply:
Perfect, thank you!